RecoTauTag/RecoTau/src/AntiElectronIDMVA6.cc

```diff
@@ -0,0 +1,965 @@
+#include "RecoTauTag/RecoTau/interface/AntiElectronIDMVA6.h"
+
+#include "FWCore/Utilities/interface/Exception.h"
+
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidate.h"
+#include "DataFormats/ParticleFlowCandidate/interface/PFCandidateFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+
+#include "CondFormats/DataRecord/interface/GBRWrapperRcd.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include <TFile.h>
+
+AntiElectronIDMVA6::AntiElectronIDMVA6(const edm::ParameterSet& cfg)
+  : isInitialized_(false),
+    mva_NoEleMatch_woGwoGSF_BL_(0),
+    mva_NoEleMatch_wGwoGSF_BL_(0),
+    mva_woGwGSF_BL_(0),
+    mva_wGwGSF_BL_(0),
+    mva_NoEleMatch_woGwoGSF_EC_(0),
+    mva_NoEleMatch_wGwoGSF_EC_(0),
+    mva_woGwGSF_EC_(0),
+    mva_wGwGSF_EC_(0)   
+{
+  loadMVAfromDB_ = cfg.exists("loadMVAfromDB") ? cfg.getParameter<bool>("loadMVAfromDB"): false;
+  if ( !loadMVAfromDB_ ) {
+    if(cfg.exists("inputFileName")){
+      inputFileName_ = cfg.getParameter<edm::FileInPath>("inputFileName");
+    }else throw cms::Exception("MVA input not defined") << "Requested to load tau MVA input from ROOT file but no file provided in cfg file";
+    
+  }
+
+  mvaName_NoEleMatch_woGwoGSF_BL_ = cfg.getParameter<std::string>("mvaName_NoEleMatch_woGwoGSF_BL");
+  mvaName_NoEleMatch_wGwoGSF_BL_ = cfg.getParameter<std::string>("mvaName_NoEleMatch_wGwoGSF_BL");
+  mvaName_woGwGSF_BL_ = cfg.getParameter<std::string>("mvaName_woGwGSF_BL");
+  mvaName_wGwGSF_BL_ = cfg.getParameter<std::string>("mvaName_wGwGSF_BL");
+  mvaName_NoEleMatch_woGwoGSF_EC_ = cfg.getParameter<std::string>("mvaName_NoEleMatch_woGwoGSF_EC");
+  mvaName_NoEleMatch_wGwoGSF_EC_ = cfg.getParameter<std::string>("mvaName_NoEleMatch_wGwoGSF_EC");
+  mvaName_woGwGSF_EC_ = cfg.getParameter<std::string>("mvaName_woGwGSF_EC");
+  mvaName_wGwGSF_EC_ = cfg.getParameter<std::string>("mvaName_wGwGSF_EC");
+
+  Var_NoEleMatch_woGwoGSF_Barrel_ = new Float_t[10];
+  Var_NoEleMatch_wGwoGSF_Barrel_ = new Float_t[18];
+  Var_woGwGSF_Barrel_ = new Float_t[24];
+  Var_wGwGSF_Barrel_ = new Float_t[32];
+  Var_NoEleMatch_woGwoGSF_Endcap_ = new Float_t[9];
+  Var_NoEleMatch_wGwoGSF_Endcap_ = new Float_t[17];
+  Var_woGwGSF_Endcap_ = new Float_t[23];
+  Var_wGwGSF_Endcap_ = new Float_t[31];
```

Comment:
More magic numbers here. Please replace them with constants. You don't even need them here if you use std::vector for Var_.